sm2/pkcs8.go

```diff
@@ -425,7 +425,7 @@ func WritePrivateKeytoPem(FileName string, key *PrivateKey, pwd []byte) (bool, e
 			Bytes: der,
 		}
 	}
-	file, err := os.Create(FileName)
+	file, err := os.OpenFile(FileName, os.O_WRONLY|os.O_CREATE, 0600)
```

Comment:
it is safe/better to create .pem file (holding private key) with 0600 permission

for example, AWS would not allow a remote ssh unless .pem file has 0600 permission